FileDialog compilation error with gtk+3

I get the following error on OpenBSD with gtk+3:
```
/tmp/audacity/lib-src/FileDialog/gtk/FileDialogPrivate.cpp:481:35: error: no matching function for call to 'gtk_file_chooser_get_current_name'
    filename = wxString::FromUTF8(gtk_file_chooser_get_current_name(m_fc));
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/gtk-3.0/gtk/gtkfilechooser.h:159:8: note: candidate function not viable: no known conversion from 'wxGtkFileChooser' to 'GtkFileChooser *' (aka '_GtkFi
leChooser *') for 1st argument
gchar *gtk_file_chooser_get_current_name (GtkFileChooser *chooser);
```

The problem has been introduced in https://github.com/audacity/audacity/commit/38bf51afea4e9838544c515c5b1cd3b0b86f0f16

--- a/lib-src/FileDialog/gtk/FileDialogPrivate.cpp
+++ b/lib-src/FileDialog/gtk/FileDialogPrivate.cpp
@@ -478,7 +478,7 @@ void FileDialog::SetFileExtension(const wxString& extension)
     wxString filename;
 
 #if defined(__WXGTK3__)
-    filename = wxString::FromUTF8(gtk_file_chooser_get_current_name(m_fc));
+    filename = wxString::FromUTF8(gtk_file_chooser_get_current_name(GTK_FILE_CHOOSER(m_widget)));
 #else
     GtkWidget *entry = find_widget(m_widget, "GtkFileChooserEntry", 0);
     if (entry)

